### DISCUSSION TITLE
Error when use Docker

### DISCUSSION THREAD

**Discussion by Tamila-2017** [open-webui/open-webui#7520] [General]
Your official Docker installation instruction which "**[GPU Support with Nvidia](https://docs.openwebui.com/getting-started/quick-start/#gpu-support-with-nvidia)**" are not working:

```
$ docker run -d -p 3000:8080 --gpus all -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:cuda
Unable to find image 'ghcr.io/open-webui/open-webui:cuda' locally
cuda: Pulling from open-webui/open-webui
2d429b9e73a6: Pull complete 
14dbff54af92: Pull complete 
71ba669986f7: Pull complete 
173289c0cbe5: Pull complete 
743a9db3af9d: Pull complete 
4f4fb700ef54: Pull complete 
cf5f90e9e1af: Pull complete 
cb12e1678f94: Pull complete 
a72e34830f66: Pull complete 
34ce2de79a35: Pull complete 
0d08235d56bf: Pull complete 
aeff68d9872a: Pull complete 
3cae3e31a5e6: Pull complete 
40987db7a16c: Pull complete 
a1901afa7ca4: Pull complete 
Digest: sha256:682d2c58154a9edddc9a03ce52026d565711e65c85e4ca52991ede094a76d775
Status: Downloaded newer image for ghcr.io/open-webui/open-webui:cuda
3c98cb56ea02582682429a157ae335e22bbc9fd0ac74fedf1a8c727d460b9f95
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].

```

- **Operating System:** 
- Debian 12.6

**Comment by panda44312** [comment]
The error message "could not select device driver "" with capabilities: [[gpu]]" usually means Docker can’t find or use your NVIDIA GPU. This might be due to a few reasons:

NVIDIA Container Toolkit Missing: Docker needs this toolkit to use GPU resources. Make sure it's installed with these commands:

```bash
sudo apt-get update
sudo apt-get install -y nvidia-container-toolkit
sudo systemctl restart docker
```

Driver Compatibility: Double-check if your NVIDIA drivers work with your GPU and the CUDA version you're using. Run nvidia-smi to make sure your driver version matches the CUDA toolkit. :)

**Comment by Tamila-2017** [comment]
Yes, of course, the NVIDIA driver is installed and works successfully with Ollama models.
Here is the output of this command's command `nvidia-smi`:

![nvidia-smi](https://github.com/user-attachments/assets/ab96acca-0fcb-4edb-8ac7-c3f5f21bc886)

**Reply by hashqueue** [reply]
> Yes, of course, the NVIDIA driver is installed and works successfully with Ollama models. Here is the output of this command's command `nvidia-smi`:
> 
> ![nvidia-smi](https://private-user-images.githubusercontent.com/27445399/391358570-ab96acca-0fcb-4edb-8ac7-c3f5f21bc886.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUxNDg5OTYsIm5iZiI6MTczNTE0ODY5NiwicGF0aCI6Ii8yNzQ0NTM5OS8zOTEzNTg1NzAtYWI5NmFjY2EtMGZjYi00ZWRiLThhYzctYzNmNWYyMWJjODg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI1VDE3NDQ1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliNTBlM2QzZGQyMmRiZjgxZGMwZmVkM2VjODlmZDNmMzczOTY2ZGQ5ZjcwMGU3MmMwZTFiYjUyMDgwNzEzMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kBDXBcU5sxque7_4_uPvo9n5_AO5e-bi2iVnCCxCkes)

see https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
